library/hls/src/main/java/com/google/android/exoplayer2/source/hls/HlsSampleStreamWrapper.java

```diff
@@ -851,11 +965,16 @@ private void initMediaChunkLoad(HlsMediaChunk chunk) {
     upstreamTrackFormat = chunk.trackFormat;
     pendingResetPositionUs = C.TIME_UNSET;
     mediaChunks.add(chunk);
-
     chunk.init(this);
-    for (HlsSampleQueue sampleQueue : sampleQueues) {
+    HlsMediaChunk loadingChunk = findChunkMatching(chunk.uid);
```

Comment:
It looks to me as if findChunkMatching will always return the same chunk as the one passed in to this method? Guess it was needed before rebasing to the latest dev branch code state.